packages/realm-network-transport/src/IterableReadableStream.ts

```diff
@@ -0,0 +1,76 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { FetchResponse, ReadableStream } from ".";
+
+const READABLE_STREAM_HANDLER: ProxyHandler<ReadableStream> = {
+  get(target, prop, receiver) {
+    if (prop === Symbol.asyncIterator) {
+      return () => {
+        const reader = target.getReader();
+        return {
+          next() {
+            return reader.read();
+          },
+          async return() {
+            await reader.cancel();
+            return { done: true, value: null };
+          },
+        };
+      };
+    }
+    return Reflect.get(target, prop, receiver);
+  },
+};
+
+/**
+ * This method ensures that a ReadableStream is iterable, wraps in a proxy with the {@link Symbol.asyncIterator} if missing.
+ * @param response
+ * @returns An async iterator.
+ */
+export function makeStreamIterable(stream: ReadableStream): ReadableStream {
+  if (Symbol.asyncIterator in stream) {
+    return stream;
+  } else if ("getReader" in stream) {
+    return new Proxy(stream, READABLE_STREAM_HANDLER);
```

Comment:
This is to gently polyfill a missing `asyncIterator` on the `ReadableStream`. We experienced with the legacy SDK that this might not always be set.